composer.json

```diff
@@ -18,6 +18,7 @@
     "require": {
         "php": "^8.2",
         "ext-zlib": "*",
+        "guzzlehttp/promises": "^2.0",
```

Comment:
We had a soft dependency on this. Now it is a hard dependency and I've added a return type into the Guzzle middleware.